A/Arb/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Arb"
+version = v"2.18.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/fredrik-johansson/arb.git", "137f3b5e13e31163634e91f8547aaae7c1ad8c32")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd arb/
+
+if [[ ${target} == *musl* ]]; then
+   export CFLAGS=-D_GNU_SOURCE=1
+   sed -i -e 's/#define _GNU_SOURCE$/#define _GNU_SOURCE 1/' thread_pool.h configure;
+fi
```

Comment:
```suggestion
elif [[ ${target} == *mingw* ]]; then
   # /lib is hardcoded in many places
   sed -i -e "s#/lib\>#/$(basename ${libdir})#g" configure
   # MSYS_NT-6.3 is not detected as MINGW
   extraflags=--build=MINGW${nbits}
fi
```
The `sed` command will adjust plenty hardcoded `/lib` entries (we could also add a patch instead) and the `--build` fixes the `.dll` suffix.